CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+      once = 1,
+      upon = 2,
+      aTime = 3
+    }
+
+    -- good
+    local thing = {
+      once = 1,
+      upon = 2,
+      aTime = 3,
+    }
+    ```
+
+    **[[back to top]](#TOC)**
+
+
+### <a name='type-coercion'>Type Casting & Coercion</a>
+
+  - Perform type coercion at the beginning of the statement. Use the built-in functions. (`tostring`, `tonumber`, etc.)
+
+  - Use `tostring` for strings if you need to cast without string concatenation.
+
+    ```lua
+    -- bad
+    local totalScore = reviewScore .. ""
+
+    -- good
+    local totalScore = tostring(reviewScore)
+    ```
+
+  - Use `tonumber` for Numbers.
+
+    ```lua
+    local inputValue = "4"
+
+    -- bad
+    local val = inputValue * 1
+
+    -- good
+    local val = tonumber(inputValue)
+    ```
+
+    **[[back to top]](#TOC)**
+
+
+### <a name='naming-conventions'>Naming Conventions</a>
+
+  - Use descriptive names. Use more descriptive names for variables with larger scopes,
+    single letter names are ok for small scopes.
+
+    ```lua
+    -- bad
+    local x = "a variable that will used through out an entire module"
+    local sum
+    for some_very_long_name = 1, 5
+      sum = sum + some_very_long_name
+    end
+    
+    -- good
+    local descriptive_name = "a variable that will used through out an entire module"
+    local sum
+    for i = 1, 5
+      sum = sum + i
+    end
+    ```
+
+  - Use underscores for ignored variables in loops or when ignoring (intermediate) return values.
+    Ignoring trailing return values with underscores is ok if it enhances clarity.
+
+    ```lua
+    --good
+    for _, name in pairs(names) do
+      -- ...stuff...
+    end
+    local result1, _, result3 = returns_three_values()
+    ```
+
+    ```lua
+    --ok
+    local some_value, _, _ = returns_three_values()
+    ```
+
+  - Use snake_case when naming objects, functions, and instances. Tend towards
+    verbosity if unsure about naming.
+
+    ```lua
+    -- bad
+    local OBJEcttsssss = {}
+    local thisIsMyObject = {}
+    local this-is-my-object = {}
+
+    local c = function()
+      -- ...stuff...
+    end
+
+    -- good
+    local this_is_my_object = {}
+
+    local function do_that_thing()
+      -- ...stuff...
+    end
+    ```
+
+  - Use PascalCase for factories.
+
+    ```lua
+    -- bad
+    local player = require('player')
+
+    -- good
+    local Player = require('player')
+    local me = Player({ name = 'Jack' })
+    ```
+
+  - Use `is` or `has` for boolean-returning functions.
+
+    ```lua
+    --bad
+    local function evil(alignment)
+      return alignment < 100
+    end
+
+    --good
+    local function is_evil(alignment)
+      return alignment < 100
+    end
+    ```
+
+    **[[back to top]](#TOC)**
+
+### <a name='modules'>Modules</a>
+
+  - The module should return a table or function.
+  - The module should not use the global namespace for anything ever.
+  - The file should be named like the module.
+
+    **[[back to top]](#TOC)**
+
```

Comment:
Some sort of closing statement here - like "Thanks for making it to the bottom! Yes, there are a lot of guidelines herein - and yes, following them is important. Everyone that uses and contributes to Kong benefits from your efforts to adhere to these guidelines, to the best of your abilities. Thanks!"